### ISSUE TITLE
Question : More Informations on electronic part

### ISSUE THREAD

**Issue by RonandoDeLaVega** [ximon/Hot-tub-remote#6]
Hello,

First of all, well done for your work, this looks amazing !!

I'm a bit new in eletronic and I can't figure out how you made your schema ? Do you have photos of actual PCB ?

Where the condensator should be linked to ?
What is the component in the middle with VCCA VCCB A B and DIR in entries ?

Thanks !
Ronando

**Comment by eddcaton** [comment]
Yes @RonandoDeLaVega  would be good to have a more in depth electrical connection guide.

**Comment by ximon** [comment]
Hi,

The esp8266 talks to the hot tub via the three pin cable between the display and the main controller.
This is done through a level shifter, a 74LVC1T45, this is the component with the VCCA,VCCB, A, B and DIR pins.

The level shifter is used as the hot tub signals are 5v and the esp8266 is 3v3, its also used to turn the output on and off before and after sending commands as its using a DMA hack which doesnt quite produce the required signals.

The condensator (capacitor) is connected across the supply just to smooth out the dips when the esp8266 is transmitting.

I've updated the schematic to clear up the net names and swap the diode with a switch to aid in local programming.

I cant upload a photo of the device as it's inside my hot tub, next time i have it open i'll take a photo.

Hope this helps!
Simon.

**Comment by jampez77** [comment]
So I've just bought some 74LVC1T45's and I was wondering 1, how on earth you connected them the anything and 2, do you know where I can get a non-mircoscopic version of them?

**Comment by Simon-Keen** [comment]
I soldered one onto [this breakout board](https://oshpark.com/shared_projects/8Vwts4jC)

You can use any non-inverting bidirectional three state driver with direction input.

**Comment by ximon** [comment]
I soldered one onto [this breakout board](https://oshpark.com/shared_projects/8Vwts4jC)

You can use any non-inverting bidirectional three state driver with direction input.

**Comment by RonandoDeLaVega** [comment]
Hello,
I found [this ](https://fr.aliexpress.com/item/32866573215.html?spm=a2g0o.productlist.0.0.e8d84650QqJwOF&algo_pvid=922f5353-e0d2-4867-9a66-0b529fdb31d6&algo_expid=922f5353-e0d2-4867-9a66-0b529fdb31d6-0&btsid=0ab6f82415898914791736417e6826&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_)which looks to be useful

**Comment by jampez77** [comment]
> Hello,
> I found [this ](https://fr.aliexpress.com/item/32866573215.html?spm=a2g0o.productlist.0.0.e8d84650QqJwOF&algo_pvid=922f5353-e0d2-4867-9a66-0b529fdb31d6&algo_expid=922f5353-e0d2-4867-9a66-0b529fdb31d6-0&btsid=0ab6f82415898914791736417e6826&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_)which looks to be useful

From what I've seen, both online and in person, the 74LVC1T45's are tiny. Smaller than the sockets in that url.

**Comment by Simon-Keen** [comment]
This is the [74LVC1T45GW SOT-363](https://uk.rs-online.com/web/p/buffer-converter-combinations/8168791/?relevancy-data=636F3D3126696E3D4931384E525353746F636B4E756D626572266C753D656E266D6D3D6D61746368616C6C26706D3D5E2828282872737C5253295B205D3F293F285C647B337D5B5C2D5C735D3F5C647B332C347D5B705061415D3F29297C283235285C647B387D7C5C647B317D5C2D5C647B377D2929292426706F3D3126736E3D592673723D2673743D52535F53544F434B5F4E554D4245522677633D4E4F4E45267573743D3831362D38373931267374613D3831363837393126&searchHistory=%7B%22enabled%22%3Atrue%7D) mounted to [this breakout](https://oshpark.com/shared_projects/8Vwts4jC)

If anybody needs one i now have two spare breakoutboards with 74LVC1T45s mounted.

It's pretty easy to solder, just tin one pad, place the part, reflow the pad, then solder the others.
Thin solder, flux and a good iron make it really easy.

![tiny](https://raw.githubusercontent.com/ximon/Hot-tub-remote/master/74LVC1T45.jpg)

**Comment by ximon** [comment]
This is the [74LVC1T45GW SOT-363](https://uk.rs-online.com/web/p/buffer-converter-combinations/8168791/?relevancy-data=636F3D3126696E3D4931384E525353746F636B4E756D626572266C753D656E266D6D3D6D61746368616C6C26706D3D5E2828282872737C5253295B205D3F293F285C647B337D5B5C2D5C735D3F5C647B332C347D5B705061415D3F29297C283235285C647B387D7C5C647B317D5C2D5C647B377D2929292426706F3D3126736E3D592673723D2673743D52535F53544F434B5F4E554D4245522677633D4E4F4E45267573743D3831362D38373931267374613D3831363837393126&searchHistory=%7B%22enabled%22%3Atrue%7D) mounted to [this breakout](https://oshpark.com/shared_projects/8Vwts4jC)

If anybody needs one i now have two spare breakoutboards with 74LVC1T45s mounted.

It's pretty easy to solder, just tin one pad, place the part, reflow the pad, then solder the others.
Thin solder, flux and a good iron make it really easy.

![tiny](https://raw.githubusercontent.com/ximon/Hot-tub-remote/master/74LVC1T45.jpg)

**Comment by RonandoDeLaVega** [comment]
Hello,

Here is my shop list, does it seems right to you ?
I've got a doubt on the capacitor : is 16V OK ?

- [switch](https://www.aliexpress.com/item/32832214619.html?spm=a2g0o.productlist.0.0.4a687fc6HYRM47&algo_pvid=764b56a4-d8fe-4269-af7f-45100355cc33&algo_expid=764b56a4-d8fe-4269-af7f-45100355cc33-0&btsid=0ab6f83115898972623797587e45e0&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_)
- [capacitor](https://www.aliexpress.com/item/32954075821.html?spm=a2g0o.productlist.0.0.28ed2175mMHiwX&algo_pvid=3f11b67e-92ef-4fde-9d57-605afba81b78&algo_expid=3f11b67e-92ef-4fde-9d57-605afba81b78-0&btsid=0ab6fab215898968040251189e0240&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_)
- [Wemos D1 Mini](https://www.aliexpress.com/item/32651747570.html?spm=a2g0s.9042311.0.0.54e04c4dNBfryS)
- [74LVC1T45 ](https://www.aliexpress.com/item/33020199381.html?spm=a2g0s.9042311.0.0.54e04c4dNBfryS)
- [SOT23 adaptator](https://www.aliexpress.com/item/33020199381.html?spm=a2g0s.9042311.0.0.54e04c4dNBfryS)

Thanks for asking ximon, but I guess sending from UK to France a breakoutboard with 74LVC1T45 mounted is more expensive than ordering from china ... 

After I'll receive all this stuff, I'll have to figure out how to flash the ESP8266 ... any keyword to search on google ? :-D

Thanks a lot !
Ronando

**Comment by jampez77** [comment]
> 
> After I'll receive all this stuff, I'll have to figure out how to flash the ESP8266 ... any keyword to search on google ? :-D
> 

Try this:  https://averagemaker.com/2018/03/wemos-d1-mini-setup.html

**Comment by ximon** [comment]
@RonandoDeLaVega  Your list seems fine apart from the sot adapter, the link is for the 74lvc1t45.
The switch is optional, you can just use a jumper if you have one spare.

You might want to check out the platformio branch too, i'm moving away from the arduino ide so all new updates will be in that branch and will eventually be merged into master.

**Comment by eddcaton** [comment]
Hey,

Can anyone tell me if this is the right part? 
https://www.ebay.co.uk/itm/193386434279

Edd

**Comment by ximon** [comment]
Hi Edd, yeah, that'll do the job

**Comment by eddcaton** [comment]
Thanks :)

**Comment by fhempy** [comment]
Hi,
I got my 74LVC1T45 today, but unfortunately I haven't found a capacitor in my spare components. Do I really need it and is it necessary to be exactly the 1000uF one you have in your semantics? I just wired it together without capacitor. Receiving signals works perfect, but sending doesn't work at all (branch: platformio-fixes).

Btw, do you use the 5V from 3-wires as power supply for ESP8266? Can you please provide more info on the power supply if you use a separate one? Many thanks!

**Comment by ximon** [comment]
The size of the capacitor isn't super critical, its just a local buffer for when the pumps turn on, you might get away without it - I added it at the time when I was seeing a lot of exceptions occurring, seemed to make a difference - but it could also have been code changes.

It might be worth hooking a scope up to the supply to the esp and watching what happens when the heater / pump turns on / off.

With regards to the platformio-fixes branch - I mistakenly checked some work in that wasn't fully tested and had to spend some time unpicking the parts that worked. I've rewrote the receiveing code which seems a lot more accurate in decoding the messages ( I tried this method before but kept seeing exceptions )

I'll check in the latest changes once i've got it tested.

**Comment by fhempy** [comment]
I just tested it with a 1000uF capacitor, unfortunately it still doesn't work. Can you please let me know if you use the 5V from the 3-wires to run the ESP8266? Receiving works perfectly, but sending doesn't work at all. It seems to send some signals as I can see all lights flashing very short on the hot tub display.

I used the platformio branch this time.

**Comment by ximon** [comment]
Yes - mine is soleley powered from the tub using the three wires going to the display controller.

Try with the main branch, i've merged the new platormio code in which runs a lot better for me, its set to run at 160MHz in the config (this might be why it wasnt working for you)

Do you have a logic analyser you can check things out with if it still doesn't work?
